adapters/openx/openx.go

```diff
@@ -140,18 +137,26 @@ func preprocess(imp *openrtb2.Imp, reqExt *openxReqExt) error {
 		imp.BidFloor = openxExt.CustomFloor
 	}
 
+	// outgoing imp.ext should be same as incoming imp.ext minus prebid and bidder
 	impExt := openxImpExt{}
-	addImpExt := false
+	if err := json.Unmarshal(imp.Ext, &impExt); err != nil {
+		return &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+	delete(impExt, "bidder")
+	delete(impExt, "prebid")
```

Comment:
This seems to be correct but there is one problem. If golang's built in delete function doesn't find "bidder" in impExt map then it will create a panic which we do not want. IMO, it would be best to first check if key exists in the map before deleting it. 
Also we should use constant already defined `openrtb_ext` package for "prebid" and "bidder", it should be PrebidExtKey and PrebidExtBidderKey respectively. 

Here is how - 
```
if _, ok := impExt[openrtb_ext.PrebidExtBidderKey]; ok {
        delete(impExt, openrtb_ext.PrebidExtBidderKey)
}

if _, ok := impExt[openrtb_ext.PrebidExtKey]; ok {
        delete(impExt, openrtb_ext.PrebidExtKey)
}
```

Reply:
> This seems to be correct but there is one problem. If golang's built in delete function doesn't find "bidder" in impExt map then it will create a panic which we do not want. IMO, it would be best to first check if key exists in the map before deleting it. Also we should use constant already defined `openrtb_ext` package for "prebid" and "bidder", it should be PrebidExtKey and PrebidExtBidderKey respectively.

Thank you for the suggestions, I will use the existing constants. Regarding the [built-in delete](https://pkg.go.dev/builtin#delete), it is now safe - turns out its behavior changed at some point to no longer need a guard and the linter will complain if I add it.

Reply:
thank you for checking that. I read it online that delete is not safe to be used on keys without checking it.